javascript/javascript1/week4.js

```diff
@@ -0,0 +1,93 @@
+//Voice assistant
+
+let userName = null;
+const todoList = [];
+function getReply(command){
+
+    if (command.toLowerCase().startsWith('hello my name is')) {
+        let name = command.substring(16).trim();
+
+        if (userName && userName.toLowerCase() === name.toLowerCase()) {
+            return `You've already introduced yourself as ${userName}.`
+        }
+        userName = name;
+        return `Nice to meet you, ${userName}.`;
+    }
+
+    else if (command.toLowerCase() === "what is my name") {
+        if (userName) {
+            return `Your name is ${userName}.`;
+        } else {
+            return "You haven't told me your name yet.";
+        }
+    }
+
+    else if (command.toLowerCase().startsWith("add") && command.toLowerCase().endsWith("to my todo")) {
+        let todoItem = command.substring(4, command.length - 11).trim();
+        todoList.push(todoItem);
+        return `${todoItem} added to your todo list.`;
+    }
+
+    else if (command.toLowerCase().startsWith("remove") && command.toLowerCase().endsWith("from my todo")) {
+        let todoItem = command.substring(7, command.length - 13).trim();
+        let itemIndex = todoList.findIndex(item => item.toLowerCase() === todoItem.toLowerCase());
+        if (itemIndex > -1) {
+            todoList.splice(itemIndex, 1);
+            return `Removed ${todoItem} from your to-do list.`;
+        } else {
+            return `${todoItem} is not in your to-do list.`;
+        }
+    }
+
+    else if (command === "What is on my todo?") {
```

Comment:
You used `command.toLowerCase()` for the previous tests. It's an excellent idea, and you should have continued with it for the rest of the exercise. It can be tedious to write it every time, so you can create a variable at the top of your function: `const cleanCommand = command.toLowerCase().trim()`